test/unit/math/rev/mat/fun/matrix_exp_action_handler_test.cpp

```diff
@@ -0,0 +1,128 @@
+#include <stan/math/rev/mat.hpp>
+#include <gtest/gtest.h>
+// #include <stan/math/prim/scal/err/check_not_nan.hpp>
+#include <test/unit/math/rev/mat/fun/util.hpp>
+#include <test/unit/math/rev/mat/util.hpp>
+#include <stan/math/rev/mat/fun/matrix_exp_action_handler.hpp>
+#include <stan/math/rev/mat/fun/to_var.hpp>
+#include <vector>
+
+TEST(MathMatrix, matrix_exp_action_diag) {
+  using MatrixType = Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>;
```

Comment:
[optional]

There are shorter ways to write double types.  Here, I'd replace this with `using Eigen::MatrixXd` and uses of `MatrixType` with `MatrixXd`.  It makes it easier for everyone else to read the code with conventional types.   We also have a Stan `matrix_d` type (and `matrix_v` and `matrix_f` and `matrix_fv` and so on for various types, plus all the vector and row vector equivalent).  

```
Eigen::MatrixXd
  == Eigen::Matrix<double, Eigen::Dynamic, Eigen::Dynamic>
  == Eigen::Matrix<double, -1, -1>
  == stan::math::matrix_d`
```

Same holds for `VectorXd` and `vector_d` and row vectors.

Reply:
fixed.
```c++
TEST(MathMatrix, matrix_exp_action_diag) {
  using Eigen::MatrixXd;
  using Eigen::VectorXd;
```